src/o365/spo/commands/file/file-get.ts

```diff
@@ -0,0 +1,277 @@
+import Auth from '../../../../Auth';
+import fs = require('fs');
+import * as path from 'path';
+import { FileProperties } from './FileProperties';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  url: string;
+  id: string;
+  asString: boolean;
+  asListItem: boolean;
+  asFile: boolean;
+  fileName: string;
+  path: string;
+}
+
+class FileGetCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FILE_GET;
+  }
+
+  public get description(): string {
+    return 'Get information about the specified file';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = (!(!args.options.id)).toString();
+    telemetryProps.url = (!(!args.options.url)).toString();
+    telemetryProps.asString = args.options.asString || false;
+    telemetryProps.asListItem = args.options.asListItem || false;
+    telemetryProps.asFile = args.options.asFile || false;
+    telemetryProps.fileName = (!(!args.options.fileName)).toString();
+    telemetryProps.path = (!(!args.options.path)).toString();
+
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let siteAccessToken: string = '';
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): request.RequestPromise => {
+        siteAccessToken = accessToken;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}. Retrieving request digest...`);
+        }
+
+        return this.getRequestDigestForSite(args.options.webUrl, siteAccessToken, cmd, this.debug);
+      })
+      .then((res: ContextInfo): request.RequestPromise => {
+        if (this.debug) {
+          cmd.log('Response:');
+          cmd.log(res);
+          cmd.log('');
+        }
+
+        if (this.verbose) {
+          cmd.log(`Retrieving file from site ${args.options.webUrl}...`);
+        }
+
+        let requestUrl: string;
+        let options: string;
+
+        if (args.options.id) {
+          requestUrl = `${args.options.webUrl}/_api/web/GetFileById('${encodeURIComponent(args.options.id)}')`;
+        }
+        else {
+          requestUrl = `${args.options.webUrl}/_api/web/GetFileByServerRelativeUrl('${encodeURIComponent(args.options.url)}')`;
+        }
+
+        if (args.options.asListItem) {
+          options = '?$expand=ListItemAllFields';
+        }
+        else if (!args.options.asListItem && !args.options.asFile && !args.options.asString) {
+          options = '';
+        }
+        else {
+          options = '/$value';
```

Comment:
It would be clearer and safer if this clause was associated explicitly with `if (args.options.asFile || args.options.asString)`

Reply:
Changed accordingly